File: sdkwork-app-sdk-java/src/main/java/com/sdkwork/app/model/ProjectUpdateForm.java
package com.sdkwork.app.model;

public class ProjectUpdateForm {
    private String projectId;
    private String projectName;
    private String projectDescription;
    private String projectIcon;
    private String projectColor;
    private List<String> projectTags;
    private ProjectSettings settings;

    public String getProjectId() {
        return this.projectId;
    }
    
    public void setProjectId(String projectId) {
        this.projectId = projectId;
    }

    public String getProjectName() {
        return this.projectName;
    }
    
    public void setProjectName(String projectName) {
        this.projectName = projectName;
    }

    public String getProjectDescription() {
        return this.projectDescription;
    }
    
    public void setProjectDescription(String projectDescription) {
        this.projectDescription = projectDescription;
    }

    public String getProjectIcon() {
        return this.projectIcon;
    }
    
    public void setProjectIcon(String projectIcon) {
        this.projectIcon = projectIcon;
    }

    public String getProjectColor() {
        return this.projectColor;
    }
    
    public void setProjectColor(String projectColor) {
        this.projectColor = projectColor;
    }

    public List<String> getProjectTags() {
        return this.projectTags;
    }
    
    public void setProjectTags(List<String> projectTags) {
        this.projectTags = projectTags;
    }

    public ProjectSettings getSettings() {
        return this.settings;
    }
    
    public void setSettings(ProjectSettings settings) {
        this.settings = settings;
    }
}

File: sdkwork-app-sdk-typescript/src/types/project-update-form.ts
import type { ProjectSettings } from './project-settings';

/** 项目更新请求 */
export interface ProjectUpdateForm {
  /** 项目ID */
  projectId?: string;
  /** 项目名称 */
  projectName?: string;
  /** 项目描述 */
  projectDescription?: string;
  /** 项目图标 */
  projectIcon?: string;
  /** 项目颜色 */
  projectColor?: string;
  /** 项目标签 */
  projectTags?: string[];
  /** 项目设置 */
  settings?: ProjectSettings;
}
